diff --git a/modules/juce_events/messages/juce_Initialisation.h b/modules/juce_events/messages/juce_Initialisation.h
index df016041a..313cd6253 100644
--- a/modules/juce_events/messages/juce_Initialisation.h
+++ b/modules/juce_events/messages/juce_Initialisation.h
@@ -123,7 +123,7 @@ public:
  #elif JUCE_ANDROID
 
   #define JUCE_CREATE_APPLICATION_DEFINE(AppClass) \
-    extern "C" juce::JUCEApplicationBase* juce_CreateApplication() { return new AppClass(); }
+    extern "C" __attribute((visibility("default"))) juce::JUCEApplicationBase* juce_CreateApplication() { return new AppClass(); }
 
   #define JUCE_MAIN_FUNCTION_DEFINITION
 
