diff --git a/modules/juce_events/messages/juce_Initialisation.h b/modules/juce_events/messages/juce_Initialisation.h
index 2fc9b96c9..975d00ea6 100644
--- a/modules/juce_events/messages/juce_Initialisation.h
+++ b/modules/juce_events/messages/juce_Initialisation.h
@@ -116,7 +116,7 @@ public:
  #elif JUCE_ANDROID
 
   #define JUCE_CREATE_APPLICATION_DEFINE(AppClass) \
-    extern "C" juce::JUCEApplicationBase* juce_CreateApplication() { return new AppClass(); }
+    extern "C" __attribute((visibility("default"))) juce::JUCEApplicationBase* juce_CreateApplication() { return new AppClass(); }
 
   #define JUCE_MAIN_FUNCTION_DEFINITION
 
