diff --git a/modules/juce_opengl/native/juce_OpenGL_android.h b/modules/juce_opengl/native/juce_OpenGL_android.h
index 96a98333af..4b6b9f5409 100644
--- a/modules/juce_opengl/native/juce_OpenGL_android.h
+++ b/modules/juce_opengl/native/juce_OpenGL_android.h
@@ -155,7 +155,7 @@ public:
                              surfaceView.get());
 
         // initialise the geometry of the view
-        updateWindowPosition (component.localAreaToGlobal (component.getLocalBounds()));
+        updateWindowPosition (component.getTopLevelComponent()->getLocalArea (&component, component.getLocalBounds()));
         hasInitialised = true;
     }
 
diff --git a/modules/juce_opengl/opengl/juce_OpenGLContext.cpp b/modules/juce_opengl/opengl/juce_OpenGLContext.cpp
index 1d32ff4cf8..7d07e599f7 100644
--- a/modules/juce_opengl/opengl/juce_OpenGLContext.cpp
+++ b/modules/juce_opengl/opengl/juce_OpenGLContext.cpp
@@ -500,7 +500,7 @@ public:
                 transform = AffineTransform::scale ((float) newArea.getWidth()  / (float) localBounds.getWidth(),
                                                     (float) newArea.getHeight() / (float) localBounds.getHeight());
 
-                nativeContext->updateWindowPosition (peer->getAreaCoveredBy (component));
+                nativeContext->updateWindowPosition (peer->getComponent().getLocalArea (&component, component.getLocalBounds()));
                 invalidateAll();
             });
         }
@@ -1127,7 +1127,7 @@ public:
                 c->handleResize();
 
             if (auto* peer = comp.getTopLevelComponent()->getPeer())
-                context.nativeContext->updateWindowPosition (peer->getAreaCoveredBy (comp));
+                context.nativeContext->updateWindowPosition (peer->getComponent().getLocalArea (&comp, comp.getLocalBounds()));
         }
     }
 
