diff --git a/modules/juce_gui_basics/native/java/app/com/rmsl/juce/ComponentPeerView.java b/modules/juce_gui_basics/native/java/app/com/rmsl/juce/ComponentPeerView.java
index 0000000000..0000000000 100644
--- a/modules/juce_gui_basics/native/java/app/com/rmsl/juce/ComponentPeerView.java
+++ b/modules/juce_gui_basics/native/java/app/com/rmsl/juce/ComponentPeerView.java
@@ -41,6 +41,7 @@ import static android.view.WindowInsetsController.BEHAVIOR_DEFAULT;
 import static android.view.WindowInsetsController.BEHAVIOR_SHOW_BARS_BY_SWIPE;
 import static android.view.WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE;
 
+import android.annotation.SuppressLint;
 import android.app.Activity;
 import android.app.Application;
 import android.content.Context;
@@ -793,6 +794,7 @@ public final class ComponentPeerView extends ViewGroup
     {
     }
 
+    @SuppressLint ("WrongConstant")
     public void setSystemUiVisibilityCompat (Window window, boolean visible, boolean isLight)
     {
         if (window != null)
