diff --git a/modules/juce_core/native/juce_Threads_android.cpp b/modules/juce_core/native/juce_Threads_android.cpp
index fe6088553..9d22f9461 100644
--- a/modules/juce_core/native/juce_Threads_android.cpp
+++ b/modules/juce_core/native/juce_Threads_android.cpp
@@ -377,9 +377,10 @@ bool Thread::createNativeThread (Priority)
             void* env = nullptr;
             androidJNIJavaVM->GetEnv (&env, JNI_VERSION_1_2);
 
+            // LAMESPACE: disabled: see the code comment on getEnv() for details.
             // only detach if we have actually been attached
-            if (env != nullptr)
-                androidJNIJavaVM->DetachCurrentThread();
+            //if (env != nullptr)
+            //    androidJNIJavaVM->DetachCurrentThread();
         }
 
         return nullptr;
