diff --git a/modules/juce_gui_basics/menus/juce_PopupMenu.cpp b/modules/juce_gui_basics/menus/juce_PopupMenu.cpp
index f70ac16b55..6305dba89f 100644
--- a/modules/juce_gui_basics/menus/juce_PopupMenu.cpp
+++ b/modules/juce_gui_basics/menus/juce_PopupMenu.cpp
@@ -378,9 +378,24 @@ struct MenuWindow final : public Component
             if (shouldDisableAccessibility)
                 setAccessible (false);

-            addToDesktop (ComponentPeer::windowIsTemporary
-                          | ComponentPeer::windowIgnoresKeyPresses
-                          | lf.getMenuWindowFlags());
-
+#if JUCE_ANDROID
+            {
+                Component* parentComp = options.getParentComponent();
+                if (parentComp == nullptr) parentComp = parent;
+                if (parentComp == nullptr) parentComp = options.getTargetComponent();
+                void* nativeHandle = nullptr;
+                if (parentComp != nullptr)
+                    if (auto* peer = parentComp->getTopLevelComponent()->getPeer())
+                        nativeHandle = peer->getNativeHandle();
+                addToDesktop (ComponentPeer::windowIsTemporary
+                              | ComponentPeer::windowIgnoresKeyPresses
+                              | lf.getMenuWindowFlags(), nativeHandle);
+            }
+#else
+            addToDesktop (ComponentPeer::windowIsTemporary
+                          | ComponentPeer::windowIgnoresKeyPresses
+                          | lf.getMenuWindowFlags());
+#endif
+
             Desktop::getInstance().addGlobalMouseListener (this);
         }

