diff -ur external/SARAH/Source/PluginProcessor.cpp external/android-audio-plugin-framework/native/juce/samples/SARAH/Source/PluginProcessor.cpp
--- Source/PluginProcessor.cpp	2020-03-18 20:56:12.396637245 +0900
+++ Source/PluginProcessor.cpp	2020-03-16 02:13:57.742597662 +0900
@@ -165,7 +165,7 @@
 
 void SARAHAudioProcessor::setStateInformation (const void* data, int sizeInBytes)
 {
-    ScopedPointer<XmlElement> xml = getXmlFromBinary(data, sizeInBytes);
+    auto xml = getXmlFromBinary(data, sizeInBytes);
     XmlElement* xprogs = xml->getFirstChildElement();
     if (xprogs->hasTagName(String("programs")))
     {
@@ -188,8 +188,8 @@
 
 void SARAHAudioProcessor::setCurrentProgramStateInformation(const void* data, int sizeInBytes)
 {
-    ScopedPointer<XmlElement> xml = getXmlFromBinary(data, sizeInBytes);
-    programBank[currentProgram].putXml(xml);
+    auto xml = getXmlFromBinary(data, sizeInBytes);
+    programBank[currentProgram].putXml(xml.get());
     sendChangeMessage();
 }
 
