diff --git a/modules/juce_opengl/native/juce_OpenGL_android.h b/modules/juce_opengl/native/juce_OpenGL_android.h
index 19f3da34a2..fd576486b0 100644
--- a/modules/juce_opengl/native/juce_OpenGL_android.h
+++ b/modules/juce_opengl/native/juce_OpenGL_android.h
@@ -109,10 +109,7 @@ public:
                              AndroidViewGroup.addView, surfaceView.get());
 
         // initialise the geometry of the view
-        auto bounds = component.getTopLevelComponent()->getLocalArea (&component, component.getLocalBounds());
-        bounds *= component.getDesktopScaleFactor();
-
-        updateWindowPosition (bounds);
+        updateWindowPosition (component.getTopLevelComponent()->getLocalArea (&component, component.getLocalBounds()));
         hasInitialised = true;
     }
 
diff --git a/modules/juce_opengl/opengl/juce_OpenGLContext.cpp b/modules/juce_opengl/opengl/juce_OpenGLContext.cpp
index fce1b76342..106e004d25 100644
--- a/modules/juce_opengl/opengl/juce_OpenGLContext.cpp
+++ b/modules/juce_opengl/opengl/juce_OpenGLContext.cpp
@@ -357,7 +357,7 @@ public:
                 transform = AffineTransform::scale ((float) newArea.getWidth()  / (float) localBounds.getWidth(),
                                                     (float) newArea.getHeight() / (float) localBounds.getHeight());
 
-                nativeContext->updateWindowPosition (peer->getAreaCoveredBy (component));
+                nativeContext->updateWindowPosition (peer->getComponent().getLocalArea (&component, component.getLocalBounds()));
 
                 if (canTriggerUpdate)
                     invalidateAll();
@@ -861,7 +861,7 @@ public:
                 c->handleResize();
 
             if (auto* peer = comp.getTopLevelComponent()->getPeer())
-                context.nativeContext->updateWindowPosition (peer->getAreaCoveredBy (comp));
+                context.nativeContext->updateWindowPosition (peer->getComponent().getLocalArea (&comp, comp.getLocalBounds()));
         }
     }
 
