diff --git a/modules/juce_gui_basics/native/juce_mac_Windowing.mm b/modules/juce_gui_basics/native/juce_mac_Windowing.mm
index a314ba95a..f28afc127 100644
--- a/modules/juce_gui_basics/native/juce_mac_Windowing.mm
+++ b/modules/juce_gui_basics/native/juce_mac_Windowing.mm
@@ -555,10 +555,10 @@ static Image createNSWindowSnapshot (NSWindow* nsWindow)
 {
     JUCE_AUTORELEASEPOOL
     {
-        CGImageRef screenShot = CGWindowListCreateImage (CGRectNull,
+        CGImageRef screenShot{};/* = CGWindowListCreateImage (CGRectNull,
                                                          kCGWindowListOptionIncludingWindow,
                                                          (CGWindowID) [nsWindow windowNumber],
-                                                         kCGWindowImageBoundsIgnoreFraming);
+                                                         kCGWindowImageBoundsIgnoreFraming);*/
 
         NSBitmapImageRep* bitmapRep = [[NSBitmapImageRep alloc] initWithCGImage: screenShot];
 
